src/libcore/macros.rs

```diff
@@ -229,14 +229,26 @@ macro_rules! try {
     })
 }
 
-/// Use the `format!` syntax to write data into a buffer.
+/// Calls `write_fmt` function on a writer
 ///
-/// This macro is typically used with a buffer of `&mut `[`Write`][write].
+/// This macro takes an implementer of [`std::fmt::Write`][fmt_write] or
+/// [`std::io::Write`][io_write] trait, a precompiled format string, and a list of arguments.
```

Comment:
The phrase 'precompiled format string' seems strange to me. The macro itself just takes a string, which can be called 'format string', but I don't see how it would be considered 'precompiled'. The thing ultimately passed to `write_fmt`, *that* I can see being called precompiled, compiled from the format string. I'd suggest removing the word 'precompiled' here.